src/components/demos/write/Demo.js

```diff
@@ -1,7 +1,143 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import styled from 'styled-components'
 
 import Header from '../../shared/Header'
+import { Context, initState } from '../../../utils/context'
+import Input from '../../shared/Input'
+import Box from '../../shared/Box'
+import { Container } from '../../shared/Common'
+
+const Wrapper = styled.div`
+  grid-area: body;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+  ${(props) => props}
+`
+const Button = styled.button((props) => props)
+const LoginAttemptText = styled.p((props) => props)
 
 export default function WriteDemo(props) {
-  return <Header {...props} showClose={true} />
+  const state = React.useContext(Context)
+  const writeDemo = { ...state.writeDemo, ...props }
+  const { update } = state
+  const { theme } = writeDemo
+
+  const [demoOutput, SetDemoOutput] = useState('')
+
+  const usernameUpperBound = 24
+  const passwordUpperBound = 16
+  const [usernameInput, SetUsernameInput] = useState('')
+  const [passwordInput, SetPasswordInput] = useState('')
+  const [someUsernameTyped, SetSomeUsernameTyped] = useState(false)
+  const [somePasswordTyped, SetSomePasswordTyped] = useState(false)
+
+  const noUsernameEntered = usernameInput.length === 0 && someUsernameTyped
+  const usernameAtMaxLength = usernameInput.length === usernameUpperBound
+  const noPasswordEntered = passwordInput.length === 0 && somePasswordTyped
+  const passwordAtMaxLength = passwordInput.length === passwordUpperBound
+
+  const UsernameErrorWarning = () => (
+    <>
+      {noUsernameEntered && <>Cannot be empty</>}
+      {usernameAtMaxLength && <>Maximum length reached</>}
+    </>
+  )
+
+  const PasswordErrorWarning = () => (
+    <>
+      {noPasswordEntered && <>Cannot be empty</>}
+      {passwordAtMaxLength && <>Maximum length reached</>}
+    </>
+  )
+
+  useEffect(() => {
+    if (usernameInput.length > 0) {
+      SetSomeUsernameTyped(true)
+    }
+    if (passwordInput.length > 0) {
+      SetSomePasswordTyped(true)
+    }
+  }, [usernameInput, passwordInput])
+
+  useEffect(() => {
+    update(initState)
+  }, [update])
+
+  const enterUsernameAndPassword = (e) => {
+    e.preventDefault()
+    SetSomeUsernameTyped(true)
+    SetSomePasswordTyped(true)
+
+    if (usernameInput.length > 0 && passwordInput.length > 0) {
+      update({
+        writeDemo: {
+          ...writeDemo,
+          usernamePasswordPairs: [
+            ...writeDemo.usernamePasswordPairs,
+            [usernameInput, passwordInput],
+          ],
+        },
+      })
+      // TODO execute API
+      SetDemoOutput('***PARSED API OUTPUT***')
```

Comment:
I think this needs some refinement. Perhaps on an unsuccessful login attempt we could display an error in this area and then on a successful login make the login screen disappear and present a desktop?